client/src/components/createCredentialDialog/createCredentialDialog.js

```diff
@@ -0,0 +1,453 @@
+              onChange={e => this.updatePassword(e)}
+            />
+          </Grid.Col>
+        </Form.FormGroup>
+      );
+    } else {
+      return (
+        <Form.FormGroup>
+          {this.renderFormLabel('SSH Key File')}
+          <Grid.Col sm={7}>
+            <Form.FormControl
+              type="text"
+              value={sshKeyFile}
+              onChange={e => this.updateSshKeyFile(e)}
+            />
+          </Grid.Col>
+        </Form.FormGroup>
+      );
+    }
+  }
+
+  renderNetworkForm() {
+    const { credentialType } = this.props;
+    const { becomeMethod, becomeUser, becomePassword } = this.state;
+
+    if (credentialType !== 'network') {
+      return null;
+    }
+
+    return (
+      <React.Fragment>
+        <Form.FormGroup>
+          <Grid.Col componentClass={Form.ControlLabel} sm={5}>
+            Become Method
+          </Grid.Col>
+          {this.renderFormLabel('SSH Key File')}
+          <Grid.Col sm={7}>
+            <div className="form-split-button">
+              <SplitButton
+                className="form-control"
+                bsStyle="default"
+                title={becomeMethod}
+                id="become-method-button"
+              >
+                {becomeMethods.map((nextMethod, index) => (
+                  <MenuItem
+                    key={index}
+                    eventKey={'become' + index}
+                    onClick={() => this.setBecomeMethod(nextMethod)}
+                  >
+                    {nextMethod}
+                  </MenuItem>
+                ))}
+              </SplitButton>
+            </div>
+          </Grid.Col>
+        </Form.FormGroup>
+        <Form.FormGroup>
+          <Grid.Col componentClass={Form.ControlLabel} sm={5}>
+            Become User
+          </Grid.Col>
+          {this.renderFormLabel('SSH Key File')}
+          <Grid.Col sm={7}>
+            <Form.FormControl
+              type="text"
+              value={becomeUser}
+              onChange={e => this.updateBecomeUser(e)}
+            />
+          </Grid.Col>
+        </Form.FormGroup>
+        <Form.FormGroup>
+          {this.renderFormLabel('Become Password')}
+          <Grid.Col sm={7}>
+            <Form.FormControl
+              type="password"
+              value={becomePassword}
+              onChange={e => this.updateBecomePassword(e)}
+            />
+          </Grid.Col>
+        </Form.FormGroup>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    const { show, credentialType } = this.props;
+    const { credentialName, authorizationType, username } = this.state;
+
+    let credentialTypeText;
+
+    switch (credentialType) {
+      case 'vcenter':
+        credentialTypeText = 'VCenter';
+        break;
+      case 'network':
+        credentialTypeText = 'Network';
+        break;
+      case 'satellite':
+        credentialTypeText = 'Satellite';
+        break;
+      default:
+        credentialTypeText = '';
+    }
```

Comment:
This setup looks related to the ```helpers.js``` work where we should probably start considering using constant objects with key/value pairs.

```js
const credentialTypeText = someGlobalConstant[credentialType] || '';
```